main/src/modules/contextualLayer/planner/src/planner.cpp

```diff
@@ -267,150 +268,213 @@ bool Planner::updateModule() {
-                        actionName,
+                    yDebug() << "recording in ABM";
+                    if (iCub->getABMClient()->Connect())
+                    {
+                        yDebug() << "ABM is connected";
+                        iCub->getABMClient()->sendActivity("reasoning",
+                            actionName,
                         "planner",  // expl: "pasar", "drives"...
                         lArgument,
                         true);
-                    yInfo() << actionName + " reasoning has been recorded in the ABM";
+                        yInfo() << actionName + " reasoning has been recorded in the ABM";
+                    }
+                    else { yInfo() << "ABMClient is not connected."; }
+
+                    // insert actions into the various holding lists (IN REVERSE ORDER)
+                    yInfo() << "storing action details:" << actionName;
+                    action_store.push_back(actionName);
+                    priority_store.push_back(priority);
+                    plan_store.push_back(planName);
+                    object_store.push_back(object);
+                    type_store.push_back(objectType);
+                    actionPos_store.push_back(ii);
 
                     if (state)
                     {
-                        stepID = ii;
                         assumption = true;
                         break;
                     }
                 }
 
                 if (assumption)
                 {
-                    // check if necessary to rank actions according to priority
-                    int insertID = 0;
-                    bool rankPriority = true;
-                    priority = command.get(1).asList()->get(1).asInt();
-                    if (priority_list.size() != 0)
-                    {
-                        for (unsigned int i = 0; i < priority_list.size(); i++)
-                        {
-                            if (priority == 1)
-                            {
-                                yInfo() << "lowest priority - append to bottom of list";
-                                rankPriority = false;
-                                break;
-                            }
-                            else if (priority < priority_list[i])
-                            {
-                                insertID = i + 1;
-                            }
-                        }
-                    }
-                    else
-                    {
-                        yInfo() << "new plan when initial state is empty";
-                        rankPriority = false;
-                    }
-
                     // insert actions into the various lists
                     if (!rankPriority)
                     {
-                        for (int i = stepID; i < grpPlans.find(planName + "-totactions").asInt() + 1; i++)
+                        for (unsigned a = action_store.size(); a-- > 0;)
```

Comment:
I used this method as it has some advantages for the iterator type `unsigned`. Here's where I got the "solution": http://stackoverflow.com/questions/4205720/iterating-over-a-vector-in-reverse-direction. The usage of unsigned integers is inherent to the function `size()`.